add clang/lld support

retrage opened the pull request:
lld, the LLVM linker, can link recent mainline Linux kernel. This PR adds lld support for lkl. The commit fa800a8 adds initial lld support for x86_64 Linux host with clang.
We will need the following features:

- [x] support more platforms
- [x] CI target for
- [x] lld support with gcc

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/lkl/linux/478)
<!-- Reviewable:end -->

tavip commented on arch/lkl/auto.conf:
```diff
@@ -1 +1,8 @@
+ifneq ($(shell $(LD) --version 2>&1 | head -n 1 | grep LLD),)
+  CC_TARGET=$(word 2,$(shell $(CC) --version | grep Target))
+  ifeq ($(CC_TARGET),x86_64-unknown-linux-gnu)
+    export OUTPUT_FORMAT=elf64-x86-64
```
Can you please rebase to latest master? I just merged a change that will require reworking this patch.

retrage replied:
I rebased and fixed the patch 2b07681.

retrage commented:
I compared the lld build with GNU ld. 
* GNU ld: 1:53.14elapsed
* LLVM lld: 1:48.28elapsed
It looks faster link time.

retrage commented:
I added i386 support and rebase the commit.

retrage commented:
I added i386 and gcc with lld build support. I also added CircleCI jobs for clang/lld both x86_64 and i386. Please refer PR to lkl/lkl-docker https://github.com/lkl/lkl-docker/pull/5.
